src/crate/theme/rtd/conf/__init__.py

```diff
@@ -186,32 +182,38 @@
     "tasklist",
 ]
 
+
 def setup(app):
-    # Force the canonical URL in multiple ways
-    #
-    # This gets around several points where the canonical_url override might be
-    # disregarded or performed out of order.
-    #
-    # This module should be star imported into `create_*.py`, and thus star
-    # imported into the base `conf.py`. Sphinx will automatically use a `setup()`
-    # in `conf.py` as an extension.
-    def force_canonical_url(app_inited):
-        from sphinx.builders.html import StandaloneHTMLBuilder
-        from sphinx.builders.epub3 import Epub3Builder
-
-        if isinstance(app_inited.builder, StandaloneHTMLBuilder) and not isinstance(
-            app_inited.builder, Epub3Builder
-        ):
-            try:
-                canonical_url = app_inited.builder.theme_options["canonical_url"]
-                canonical_url_path = app_inited.builder.theme_options[
-                    "canonical_url_path"
-                ]
-            except KeyError:
-                return
-            canonical_url = canonical_url + canonical_url_path
-            app_inited.env.config.html_context["canonical_url"] = canonical_url
-            app_inited.builder.config.html_context["canonical_url"] = canonical_url
+
+    # Configure Sphinx/RTD to host projects on a custom domain, but also on a non-root resource.
+    def configure_self_hosted_on_path(app_inited):
+        """
+        By default, Sphinx is agnostic, and RTD only supports custom domains,
+        without support for hosting on non-root resources.
+
+        In order to get things in order, this routine configures:
+
+        - `ogp_site_url` Sphinx setting
+          Configures the `sphinxext.opengraph` Sphinx addon.
+
+        - `custom_baseurl` HTML context variable
+          Configures the version chooser defined in `version_chooser.html`.
+        """
+
+        config = app_inited.env.config
+        html_baseurl = config.html_baseurl
+
+        # Resolve an intertwingulation between vanilla Sphinx and RTD.
+        # CrateDB docs are hosted on a custom domain behind Nginx.
+        html_baseurl_real = html_baseurl + "en/latest/"
+
+        # Configure `sphinxext.opengraph`.
+        config.ogp_site_url = html_baseurl_real
+
+        # Properly render rel="canonical" links.
+        # Because RTD does something silly with `index` page names,
+        # render it in `base.html` manually.
+        config.html_context["custom_baseurl"] = html_baseurl_real
```

Comment:
That's the new project-wide configuration, so that relevant settings do not need to be configured individually on a per-project level.